cdp-agentkit-core/cdp_agentkit_core/actions/wow/buy_token.py

```diff
@@ -0,0 +1,76 @@
+from collections.abc import Callable
+
+from cdp import Wallet
+from pydantic import BaseModel, Field
+
+from cdp_agentkit_core.actions import CdpAction
+from cdp_agentkit_core.actions.wow.constants import (
+    WOW_ABI,
+)
+from cdp_agentkit_core.actions.wow.quotes import get_buy_quote
+from cdp_agentkit_core.actions.wow.uniswap.index import get_has_graduated
+
+WOW_BUY_TOKEN_PROMPT = """
+This tool will buy a Zora Wow ERC20 memecoin with ETH. This tool takes the WOW token contract address, the address to receive the tokens, and the amount of ETH to spend (in wei, meaning "1" is 1 wei or 0.000000000000000001 of ETH). The amount is a string and cannot have any decimal points, since the unit of measurement is wei. Make sure to use the exact amount provided, and if there's any doubt, check by getting more information before continuing with the action. The minimum to buy is 100000000000000 wei which is 0.0000001 ether. It is only supported on Base Sepolia and Base Mainnet.
+"""
+
+
+class WowBuyTokenInput(BaseModel):
+    """Input argument schema for buy token action."""
+
+    contract_address: str = Field(
+        ...,
+        description="The WOW token contract address, such as `0x036CbD53842c5426634e7929541eC2318f3dCF7e`",
+    )
+
+    amount_eth_in_wei: str = Field(
+        ...,
+        description="Amount of ETH to spend (in wei), meaning 1 is 1 wei or 0.000000000000000001 of ETH",
+    )
+
+
+def wow_buy_token(wallet: Wallet, contract_address: str, amount_eth_in_wei: str) -> str:
+    """Buy a Zora Wow ERC20 memecoin with ETH.
+
+    Args:
+        wallet (Wallet): The wallet to create the token from.
+        contract_address (str): The WOW token contract address, such as `0x036CbD53842c5426634e7929541eC2318f3dCF7e`
+        amount_eth_in_wei (str): Amount of ETH to spend (in wei), meaning 1 is 1 wei or 0.000000000000000001 of ETH
+
+    Returns:
+        str: A message containing the token purchase details.
+
+    """
+    token_quote = get_buy_quote(wallet.network_id, contract_address, amount_eth_in_wei)
+
+    # Multiply by 99/100 and floor to get 99% of quote as minimum
+    min_tokens = str(int((token_quote * 99) // 100))  # Using integer division to floor the result
+
+    has_graduated = get_has_graduated(wallet.network_id, contract_address)
+    invocation = wallet.invoke_contract(
+        contract_address=contract_address,
+        method="buy",
+        abi=WOW_ABI,
+        args={
+            "recipient": wallet.default_address.address_id,
+            "refundRecipient": wallet.default_address.address_id,
+            "orderReferrer": "0x0000000000000000000000000000000000000000",
+            "expectedMarketType": has_graduated and "1" or "0",
+            "minOrderSize": min_tokens,
+            "sqrtPriceLimitX96": "0",  # TODO
```

Comment:
Can we add a more descriptive TODO here.

Reply:
removed the TODO - double checked with zora in that thread that 0 is fine